### ISSUE TITLE
Model.track probably causing stream to display frames slowly

### ISSUE THREAD

**Issue by DokterPP** [ultralytics/ultralytics#3908]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hi, I'm trying to make a program to detect a label and I recently trained a new yolov8 model. When training, I trained it using the latest Ultralytics version on Colab causing me to update the version on my own environment too. Previously I had an older model which worked fine and the RTSP feed was working smoothly in real time. Now using the exact same code but with the new Ultralytics version, the stream is smooth but however, doesnt skip frames to try and keep up to real-time. I understand a delay is expected but it gets slower and slower until the stream pauses and has to jump to real-time; about a few minutes ahead. This means I lose a lot of footage that goes unseen. 
the error is ( h264 error while decoding MB 83 39, bytestream -6)
and it will also say (h264 left block unavailable for requested intra4X4 mode -1)

```
 model = YOLO("models/v8s16.pt")
    CLASS_NAMES_DICT = model.model.names
    for result in model.track(source='rtsp://stream', stream=True, agnostic_nms=True):
        frame = result.orig_img
```
        
The rest of the code I haven't included is just loops and I have run the code without those loops and the issue still persists.

### Environment

ultralytics==8.0.138
torch==2.0.1+cu118
Python 3.9.6

### Minimal Reproducible Example

```
import cv2
from ultralytics import YOLO

def main():
    
    model = YOLO("models/v8s16.pt")
    for result in model.track(source='rtsp_stream', stream=True, agnostic_nms=True, persist=True):
        frame = result.orig_img
        cv2.imshow("yolov8", frame)

        if (cv2.waitKey(1) & 0xFF == ord('q')):
            break
    cv2.destroyAllWindows()

if _name_ == "_main_":

    main()
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@DokterPP Hi, I just tested yolov8s tracking with the same script as yours for like 6mins, but the inference runs smoothly all the time. I'm not able to reproduce the issue.

**Comment by DokterPP** [comment]
> @DokterPP Hi, I just tested yolov8s tracking with the same script as yours for like 6mins, but the inference runs smoothly all the time. I'm not able to reproduce the issue.

DId you see if its the same ultralytics version?

here are my pip installations for ur reference

aggdraw==1.3.15
atomicwrites==1.4.0
attrs==21.2.0
backcall==0.2.0
certifi==2021.10.8
charset-normalizer==2.0.11
click==8.1.3
cloudpickle==2.2.0
colorama==0.4.6
customtkinter==5.1.2
cycler==0.11.0
darkdetect==0.8.0
dask==2022.11.0
debugpy==1.5.1
decorator==5.1.0
entrypoints==0.3
filelock==3.10.0
Flask==2.2.2
Flask-SQLAlchemy==3.0.2
Flask-WTF==1.0.1
fonttools==4.28.2
fsspec==2022.11.0
gitdb==4.0.10
GitPython==3.1.31
greenlet==2.0.1
gym==0.23.0
gym-notices==0.0.8
HeapDict==1.0.1
idna==3.3
importlib-metadata==5.1.0
iniconfig==1.1.1
install==1.3.5
ipykernel==6.6.0
ipython==7.30.1
itsdangerous==2.1.2
jedi==0.18.1
Jinja2==3.1.2
joblib==1.1.0
jupyter-client==7.1.0
jupyter-core==4.9.1
keyboard==0.13.5
kiwisolver==1.3.2
lap==0.4.0
llvmlite==0.39.1
locket==1.0.0
MarkupSafe==2.1.1
matplotlib==3.5.0
matplotlib-inline==0.1.3
mpmath==1.3.0
msgpack==1.0.4
multipledispatch==0.6.0
mysql-connector-python==8.0.27
nest-asyncio==1.5.4
networkx==3.0
numba==0.56.4
numpy==1.24.2
opencv-python==4.7.0.72
package-name==0.1
packaging==21.2
pandas==1.5.1
parso==0.8.3
partd==1.3.0
pickleshare==0.7.5
Pillow==8.4.0
plotly==5.5.0
pluggy==1.0.0
prompt-toolkit==3.0.24
psutil==5.9.4
py==1.10.0
pygame==2.1.0
Pygments==2.11.1
pyodbc==4.0.35
pyparsing==2.4.7
pytest==6.2.5
python-dateutil==2.8.2
pytz==2022.6
pywin32==303
PyYAML==6.0
pyzmq==22.3.0
requests==2.27.1
scikit-learn==1.1.2
scipy==1.9.0
seaborn==0.12.2
sentry-sdk==1.17.0
setuptools-scm==6.3.2
six==1.16.0
sklearn==0.0.post1
smmap==5.0.0
sortedcontainers==2.4.0
SQLAlchemy==1.4.44
sqlparse==0.4.3
squarify==0.4.3
supervision==0.3.0
sympy==1.11.1
tbb==2021.7.1
tblib==1.7.0
tenacity==8.0.1
thop==0.1.1.post2209072238
threadpoolctl==3.1.0
tifffile==2023.1.23.1
toml==0.10.2
tomli==1.2.2
toolz==0.12.0
torch==2.0.1+cu118
torchaudio==2.0.2+cu118
torchvision==0.15.2+cu118
tornado==6.1
tqdm==4.65.0
traitlets==5.1.1
typing-extensions==4.4.0
tzdata==2022.7
ultralytics==8.0.138
urllib3==1.26.15
utils==1.0.1
visualkeras==0.0.2
wcwidth==0.2.5
Werkzeug==2.2.2
WTForms==3.0.1
yellowbrick==1.4
zict==2.2.0
zipp==3.11.0

**Comment by DokterPP** [comment]
@Laughing-q  could you quote your modules such that i can look into the dependencies

**Comment by Laughing-q** [comment]
@DokterPP I'm using the latest version 8.0.141
